tasks/task_16/service/service.go

```diff
@@ -0,0 +1,70 @@
+package service
+
+import (
+	"context"
+	"fmt"
+
+	"studing/tasks/task_16/models"
+)
+
+type Service struct {
+	scanner scanner
+	printer printer
+}
+
+func New(s scanner, p printer) *Service {
+	return &Service{
+		scanner: s,
+		printer: p,
+	}
+}
+
+type printer interface {
+	Print(ctx context.Context, excelFile *models.ExcelFile) error
+}
+
+type source interface {
+	GetExcelFiles(ctx context.Context) ([]*models.ExcelFile, error)
+	GetFileByName(ctx context.Context, name string) (*models.ExcelFile, error)
```

Comment:
"GetFileByName" this method is useless, you can select all files by "GetExcelFiles"